cfd_protocol/src/protocol/transactions.rs

```diff
@@ -21,6 +21,45 @@ use std::num::NonZeroU8;
 /// In satoshi per vbyte.
 const SATS_PER_VBYTE: f64 = 1.0;
 
+#[derive(Clone, Debug)]
+struct Fee {
+    fee: f64,
+}
+
+impl Fee {
+    pub fn new(signed_vbytes: f64) -> Self {
+        let fee = signed_vbytes * SATS_PER_VBYTE;
+        Self { fee }
+    }
+
+    /// Note: does not mutate, but creates another Fee object
```

Comment:
You could put `#[must_use]` here so the compiler generates a warning if you don't use the result.

Alternatively, consuming `self` will also make this harder to misuse.